scripts/explain.js

```diff
@@ -0,0 +1,137 @@
+// Description:
+//   A place to get initial explanation of products or concepts at Canonical that reads from a Google spreadsheet.
+//
+// Dependencies:
+//   google-spreadsheet: ""
+//
+// Configuration:
+//   set HUBOT_SPREADSHEET_ID in environment
+//   set HUBOT_SPREADSHEET_CLIENT_EMAIL in environment
+//   set HUBOT_SPREADSHEET_PRIVATE_KEY in environment
+//   set MATTERMOST_TOKEN_CMD_ACRONYM in environment
+//
+// Commands:
+//   hubot explain <product|concept>: explains the concepts and provides reading resources
+//
+// URLS:
+//   POST /hubot/explain
+//     Follows format suggested here: https://docs.mattermost.com/developer/slash-commands.html
+//     data:
+//       token: Should be similar than MATTERMOST_TOKEN_CMD_ACRONYM
+//       text: explain
+//     response:
+//       {"response_type": "ephemeral", "text": TEXT_POSTED}
+//
+// Note:
+//   The format of the spreadsheet should be:
+//
+//   | Explain | Definition                            | PM          | Contact | Link                                                                                              |
+//   | MAAS    | MAAS is a fast provisioning tool..... | Anton Smith | ~MAAS   | https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests |
+//
+// Authors:
+//   tbille, amylily1011
+
+var SPREADSHEET_ID = process.env.HUBOT_SPREADSHEET_ID;
+if (!SPREADSHEET_ID) {
+    console.log("Missing HUBOT_SPREADSHEET_ID in environment");
+}
+
+var CLIENT_EMAIL = process.env.HUBOT_SPREADSHEET_CLIENT_EMAIL;
+if (!CLIENT_EMAIL) {
+    console.log("Missing HUBOT_SPREADSHEET_CLIENT_EMAIL in environment");
+}
+
+var PRIVATE_KEY = process.env.HUBOT_SPREADSHEET_PRIVATE_KEY;
+if (!PRIVATE_KEY) {
+    console.log('Missing HUBOT_SPREADSHEET_PRIVATE_KEY in environment');
+} else {
+    PRIVATE_KEY = PRIVATE_KEY.replace(/\\n/g, '\n');
+}
+
+var CREDS = {
+    client_email: CLIENT_EMAIL,
+    private_key: PRIVATE_KEY
+};
+
+const { GoogleSpreadsheet } = require('google-spreadsheet');
+const doc = new GoogleSpreadsheet(SPREADSHEET_ID);
+
+var HTTPS_PROXY = process.env.HTTPS_PROXY;
+if (HTTPS_PROXY) {
+    doc.axios.defaults.proxy = false;
+    const HttpsProxyAgent = require('https-proxy-agent');
+    doc.axios.defaults.httpsAgent = new HttpsProxyAgent(HTTPS_PROXY);
+}
+
+var MATTERMOST_TOKEN_CMD_ACRONYM = process.env.MATTERMOST_TOKEN_CMD_ACRONYM;
+if (!MATTERMOST_TOKEN_CMD_ACRONYM) {
+    console.log("Missing MATTERMOST_TOKEN_CMD_ACRONYM in environment");
+}
+
+async function googleSpreadsheetHandler(explain) {
+    var explain;
+    await doc.useServiceAccountAuth(CREDS);
+    await doc.loadInfo();
+    const sheet = doc.sheetsByTitle['Explain'];
+    const sheet_why = doc.sheetsByTitle['Why'];
+    var rows = await sheet.getRows();
+
+    //get 'why' rows from sheet['Why']
```

Comment:
Shouldn't we do this operation once we know the param is `why`. So within the if statement below which returns.